contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,124 @@
-    bytes32 _grantRegistryId,
-    string memory _metaPtr
-  ) {
-    owner = _owner;
-    //state = State.CONFIG;
-    grantRegistryId = _grantRegistryId;
-    metaPtr = _metaPtr;
-  }
-
-  // State
-  enum State {
-    CONFIG,
-    ACTIVE,
-    CLOSED
-  }
-  //State public state;
-
-  // Grant Pointer Object
-  // Mapping from GrantID to set of Owner Address, Payout Address, Metadata Pointer, Replaces Grant (optional)
-  // TODO: Implement safemath for avoiding overflows with ints
-  uint256 public grantCount = 0;
-  mapping(bytes32 => Grant) public grants;
-  bytes32[] public grantIds;
+  // --- Data ---
+  /// @notice Number of grants stored in this registry
+  uint96 public grantCount;
 
+  /// @notice Grant object
   struct Grant {
-    bytes32 id;
-    address owner;
-    address payout;
-    string metaPtr;
-    State state;
-    bytes32 replaces;
+    uint96 id; // grant ID, as uint96 to pack into same slot as owner (this implies a max of 2^96-1 = 7.9e28 grants)
+    address owner; // grant owner (has permissions to modify grant information)
+    address payee; // address that receives funds donated to this grant
+    string metaPtr; // URL pointing to grant metadata (for off-chain use)
   }
 
-  function activate(bytes32 _id) external onlyOwner {
-    grants[_id].state = State.ACTIVE;
-  }
+  /// @notice Mapping from Grant ID to grant data
+  mapping(uint96 => Grant) public grants; // TODO use an array instead with ID to index it? Which is better? Will array copy full array to memory in `getAllGrants`?
 
-  function closed(bytes32 _id) public onlyOwner {
-    grants[_id].state = State.CLOSED;
-  }
+  // --- Events ---
+  /// @notice Emitted when a new grant is created
+  event GrantCreated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
+
+  /// @notice Emitted when a grant's owner is changed
+  event GrantUpdated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
 
-  function isConfig(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.CONFIG;
+  // --- Core methods ---
+  /**
+   * @notice Create a new grant in the registry
+   * @param _owner Grant owner (has permissions to modify grant information)
+   * @param _payee Address that receives funds donated to this grant
+   * @param _metaPtr URL pointing to grant metadata (for off-chain use)
+   */
+  function createGrant(
+    address _owner,
+    address _payee,
+    string memory _metaPtr
+  ) external {
+    uint96 _id = grantCount;
+    grants[_id] = Grant(_id, _owner, _payee, _metaPtr);
+    emit GrantCreated(_id, _owner, _payee, _metaPtr);
+    grantCount += 1;
   }
 
-  function isActive(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.ACTIVE;
+  /**
+   * @notice Update the owner of a grant
+   * @param _id ID of grant to update
+   * @param _owner New owner address
+   */
+  function updateGrantOwner(uint96 _id, address _owner) external {
```

Comment:
Can you expand on your reasoning for 4 separate update methods, instead of just 1? Is it just gas savings?

Reply:
Yea the two reasons I thought of for splitting this up are:

1. Gas savings: The separate methods ensures you only write to a minimum number of storage slots
2. UX/devex: Requiring all of `owner`, `payee`, and `metaPtr` to be passed just to update one of those 3 is inconvenient and error prone

Reply:
Sounds good. I'm fine with keeping all four, though it's worth considering that if you want to update 2 of 3 params, you still end up in the same situation gas/DevEx wise. One thing we might want to discuss is whether a multicall method on the contract might be helpful in this regard.

Reply:
Yea I thought about adding a multicall but figured this probably wasn't the right PR for it. Created https://github.com/dcgtc/dgrants/issues/9 to track it